apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/preview/@modal/(.)triggers/[triggerUuid]/edit/EditTriggerModal.tsx

```diff
@@ -48,8 +56,13 @@ function useDocumentTrigger({ triggerUuid }: { triggerUuid: string }) {
     [project.id, commit.uuid, isHead],
   )
   const onCloseModal = useCallback(() => {
-    navigate.push(previewPath, { scroll: false })
-  }, [navigate, previewPath])
+    if (isOnboarding) {
```

Comment:
smelly API, it means consumers of this hook know of its internal implementation (tight coupling). You should pass a full onCloseModal callback as a prop of this hook and define this callback differently on each consumer.